lib/logstash/inputs/gelf.rb

```diff
@@ -142,6 +145,29 @@ def self.coerce_timestamp(timestamp)
     timestamp.is_a?(BigDecimal) ? LogStash::Timestamp.at(timestamp.to_i, timestamp.frac * 1000000) : LogStash::Timestamp.at(timestamp)
   end
 
+  # from_json_parse uses the Event#from_json method to deserialize and directly produce events
+  def self.from_json_parse(json)
+    LogStash::Event.from_json(json).each { |event| event }
+  rescue LogStash::Json::ParserError => e
+    logger.error("JSON parse failure. Falling back to plain-text", :error => e, :data => json)
+    LogStash::Event.new(MESSAGE_FIELD => json, TAGS_FIELD => [PARSE_FAILURE_TAG])
+  end # def self.from_json_parse
+
+  # legacy_parse uses the LogStash::Json class to deserialize json
+  def self.legacy_parse(json)
+    o = LogStash::Json.load(json)
+    LogStash::Event.new(o) if o
+  rescue LogStash::Json::ParserError => e
+    logger.error("JSON parse failure. Falling back to plain-text", :error => e, :data => json)
```

Comment:
since we have the exact same text twice, we should define a constant here too IMO